homeassistant/components/flipr/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Sensor platform for the Flipr's pool_sensor."""
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import FliprEntity
+from .const import ATTRIBUTION, CONF_FLIPR_ID, DOMAIN, MANUFACTURER, NAME
+
+SENSORS = {
+    "chlorine": {
+        "unit": "mV",
+        "icon": "mdi:pool",
+        "name": "Chlorine",
+        "device_class": None,
+    },
+    "ph": {"unit": None, "icon": "mdi:pool", "name": "pH", "device_class": None},
+    "temperature": {
+        "unit": TEMP_CELSIUS,
+        "icon": None,
+        "name": "Water Temp",
+        "device_class": DEVICE_CLASS_TEMPERATURE,
+    },
+    "date_time": {
+        "unit": None,
+        "icon": None,
+        "name": "Last Measured",
+        "device_class": DEVICE_CLASS_TIMESTAMP,
+    },
+    "red_ox": {
+        "unit": "mV",
+        "icon": "mdi:pool",
+        "name": "Red OX",
+        "device_class": None,
+    },
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors_list = []
+    for sensor in SENSORS:
+        sensors_list.append(
+            FliprSensor(coordinator, config_entry.data[CONF_FLIPR_ID], sensor)
+        )
+
+    async_add_entities(sensors_list, True)
+
+
+class FliprSensor(FliprEntity, Entity):
+    """Sensor representing FliprSensor data."""
+
+    @property
+    def device_info(self):
+        """Define device information global to entities."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.flipr_id)
+            },
+            "name": NAME,
+            "manufacturer": MANUFACTURER,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the particular component."""
+        return f"Flipr {self.flipr_id} {SENSORS[self.info_type]['name']}"
```

Comment:
Would be good to have a core maintainer comment on this one. Since we have the device registry, different physical devices are already divided by Flipr id, thus do we still need the name of the integration and name of the device in every entity.

I can imagine that this is most readable one, but I wonder what the current guidance is. (also for other integrations).

Reply:
Yes I prefer to have the "flipr" prefix because otherwise, when you configure lovelace content, it is quite harder to find the entities...
I checked my global entity list and found that for my samsung phones, all entities are prefixed by this "sm_[model_number]".

Reply:
>For name and entity_id I don't think we have a standard in that regard. I usually go with what the contributor suggests as name.

Let's keep your suggestions ;-).

Reply:
OK, cool.